api-2.2/src/test/java/org/openmrs/htmlformentryui/element/ConditionElementTest.java

```diff
@@ -0,0 +1,194 @@
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.springframework.mock.web.MockHttpServletRequest;
+
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(Context.class)
+public class ConditionElementTest {
+	
+	private ConditionElement element;
+	private MockHttpServletRequest request;
+	private final String CONDITION_NAME_WIDGET_ID = "condition-name";
+	private final String NONE_CODED_CONCEPT = "Test None Coded Concept";
+	private final String CONDITION_REQUIRED_ERROR = "Condition required";
+	private final String END_DATE_BEFORE_ONSET_DATE_ERROR = "End date can't be before onset date";
+	@Mock
+    private MessageSourceService messageSourceService;
+	@Mock
+	private ConditionService conditionService;
+	@Mock
+	private FormEntrySession session;
+	@Mock
+	private FormEntryContext context;
+	@Mock
+	private ConceptSearchAutocompleteWidget conditionSearchWidget;
+	@Mock
+	private RadioButtonsWidget conditionStatusWidget;
+	@Mock
+	private DateWidget endDate;
+	@Mock
+	private DateWidget onsetDate;
+
+	@Before
+	public void setup() {
+		// Stub services
+		when(conditionService.saveCondition(any(Condition.class))).thenAnswer(new Answer<Condition>() {
+			  @Override
+			  public Condition answer(InvocationOnMock invocation) throws Throwable {
+			    return (Condition) invocation.getArguments()[0];
+			  }
+			});
+		
+		mockStatic(Context.class);
+		when(Context.getConditionService()).thenReturn(conditionService);
+		when(Context.getMessageSourceService()).thenReturn(messageSourceService);
+		
+		// Setup html form session context
+		when(context.getMode()).thenReturn(Mode.ENTER);
+		request = new MockHttpServletRequest();
+		when(session.getContext()).thenReturn(context);
+		when(session.getPatient()).thenReturn(new Patient(1));
+		
+		when(onsetDate.getValue(context, request)).thenReturn(new GregorianCalendar(2014, Calendar.FEBRUARY, 11).getTime());
+		when(endDate.getValue(context, request)).thenReturn(new GregorianCalendar(2018, Calendar.DECEMBER, 1).getTime());
+		
+		// setup condition element
+		element = spy(ConditionElement.class);
+		element.setConditionSearch(conditionSearchWidget);
+		element.setConditionStatusWidget(conditionStatusWidget);
+		element.setOnSetDate(onsetDate);
+		element.setEndDate(endDate);
+		
+	}
+	
+	@Test
+	public void handleSubmission_shouldCreateNewCondition() {
+		// setup
+		when(conditionSearchWidget.getValue(context, request)).thenReturn("1519");
+		when(conditionStatusWidget.getValue(context, request)).thenReturn("active");
+		
+		// replay
+		element.handleSubmission(session, request);
+		
+		// verify
+		ArgumentCaptor<Condition> captor = ArgumentCaptor.forClass(Condition.class);
+		verify(conditionService).saveCondition(captor.capture());
+		Condition condition = captor.getValue();
+		Assert.assertEquals(ConditionClinicalStatus.ACTIVE, condition.getClinicalStatus());
+		Assert.assertNotNull(condition.getPatient());
+		Assert.assertNotNull(condition.getCondition());		
```

Comment:
Is there no way here to verify that the condition is indeed based on concept 1519?